red.py

```diff
@@ -174,6 +176,32 @@ def user_allowed(self, message):
         else:
             return True
 
+    async def pip_install(self, name, *, timeout=60):
```

Comment:
could prompt the user when it takes too long. ask them to keep trying or kill it